### PR TITLE
[PPP-5459][Create common-ui Utilities]

### PR DESCRIPTION
**Pull request by mmujahiduddin**
Create common-ui Utilities

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 10m 40s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5236**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1820/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: impl/client/src/main/javascript/web/util/xss.js [hunk 1]
```diff
@@ -125,23 +133,36 @@ define("common-ui/util/xss", ["common-ui/dompurify"], function(DOMPurify) {
      *
      * Uses {@link #sanitizeHtml} to sanitize the given, possibly unsafe, HTML text.
      *
-     * @param {Element} elem - The HTML element.
+     * @param {HTMLElement|jQuery} elem - The HTML element.
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/util/xss.js:136]:
```
Set's an element or jQuery object's inner HTML ...

....

@param {HTMLElement|jQuery} elem - The HTML element or jQuery object.
```

File: impl/client/src/main/javascript/web/util/xss.js [hunk 2]
```diff
@@ -125,23 +133,36 @@ define("common-ui/util/xss", ["common-ui/dompurify"], function(DOMPurify) {
...
+      } else {
+        throw new Error("Invalid argument 'elem'.");
+      }
     },
 
     /**
      * Set's an element's HTML to a given text, without sanitizing the text.
      *
-     * @param {Element} elem - The HTML element.
+     * @param {HTMLElement} elem - The HTML element.
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/util/xss.js:154]: Same issue as for `setHtml`

File: impl/client/src/test/javascript/util/xss.spec.js [hunk 3]
```diff
@@ -76,6 +76,19 @@ define(["common-ui/util/xss"], function(xssUtil) {
           xssUtil.setHtml(elem, unsafeHtml);
           expect(elem.innerHTML).not.toContain("<script>");
         });
+
+        it("should set sanitized HTML to a jQuery object", function () {
+          const elem = $("<div></div>");
```
Comment (dcleao) [inline impl/client/src/test/javascript/util/xss.spec.js:81]: Should there not be a `"common-ui/jquery"` dependency in this (test) module, with parameter `$`?

File: impl/client/src/test/javascript/util/xss.spec.js [hunk 4]
```diff
@@ -151,5 +177,111 @@ define(["common-ui/util/xss"], function(xssUtil) {
...
-  });
+    describe("open", function() {
+      it("should open a new window with a sanitized HTTP URL", function() {
+        const url = "http://example.com";
+        const windowOpenSpy = spyOn(window, "open");
+        xssUtil.open(url);
+        expect(windowOpenSpy).toHaveBeenCalledWith("http://example.com/");
+      });
+
+      it("should open a new window with a sanitized HTTPS URL", function() {
```
Comment (dcleao) [inline impl/client/src/test/javascript/util/xss.spec.js:188]: Missing a test proving an unsafe URL gets sanitized.

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 11m 6s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5251**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1820/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 11m 21s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5246**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1820/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]
💯

**Pull request merged**